src/main/com/smithsmodding/armory/common/factories/ArmorFactory.java

```diff
@@ -35,69 +37,141 @@ public static ArmorFactory getInstance() {
 
         if (!validateOldAgainstNewAddons(existingExtensions, newAddons))
             throw new IllegalArgumentException("ADDONS not compatible");
 
-        IMultiComponentArmorCapability existingCapability = baseStack.getCapability(ModCapabilities.MOD_MULTICOMPONENTARMOR_CAPABILITY, null);
-        ItemStack combinedArmor = buildNewMLAArmor(armor, compressInformation(existingExtensions, newAddons), newTotalDurability, coreMaterial);
+        final List<IMultiComponentArmorExtensionInformation> combinedExtensions = compressInformation(existingExtensions, newAddons);
 
+        final int newMaximalDurability =
+          ArmorHelper.getModifyableCapabilityValue(armor, coreMaterial, combinedExtensions, ModCapabilities.MOD_ARMOR_DURABILITY_CAPABILITY).intValue();
+        if (newMaximalDurability <= 0)
+        {
+            throw new IllegalArgumentException(String.format("Combination of armor, material and extensions results in a <= 0 durability: %s,%s, %s",
+              armor,
+              coreMaterial,
+              combinedExtensions));
+        }
+
+        final int newCurrentDurability = newMaximalDurability - currentUsedDurability;
+
+        ItemStack combinedArmor = buildNewMLAArmor(armor, coreMaterial, combinedExtensions);
         IMultiComponentArmorCapability newCapability = combinedArmor.getCapability(ModCapabilities.MOD_MULTICOMPONENTARMOR_CAPABILITY, null);
-        newCapability.setCurrentDurability(existingCapability.getCurrentDurability() + (newTotalDurability - existingCapability.getMaximalDurability()));
+
+        newCapability.setCurrentDurability(newCurrentDurability);
+        setArmorCustomName(baseStack, currentCustomName);
 
         return combinedArmor;
     }
 
     /**
      * Function used to create a new armor ItemStack
      *
-     * @param armor           The base armor used to create the new armor.
-     * @param addons          The new addons stored in a ArrayList
-     * @param totalDurability
-     * @param coreMaterial
-     * @param data            Extra data for your factories  @return A new ItemStack with full getDurability
+     * @param armor The base armor used to create the new armor.
+     * @param coreMaterial The core material of the armor.
+     * @param addons    The new addons stored in a ArrayList
+     * @return A new ItemStack with full durability.
+     * @throws IllegalArgumentException When the Addons are not compatible, or when the durability is smaller or equal to 0
      */
     @Nonnull
     @Override
-    public ItemStack buildNewMLAArmor(IMultiComponentArmor armor, ArrayList<IMultiComponentArmorExtensionInformation> addons, Integer totalDurability, ICoreArmorMaterial coreMaterial, Object... data) throws IllegalArgumentException {
+    public ItemStack buildNewMLAArmor(IMultiComponentArmor armor, ICoreArmorMaterial coreMaterial, List<IMultiComponentArmorExtensionInformation> addons)
+      throws IllegalArgumentException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)